// valve-kv/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createDuplicateKeyArray = exports.arrayFromKvObject = exports.arrayToKvObject = exports.isDuplicateKeyArray = exports.isKvObject = exports.deserializeFile = exports.deserialize = exports.serialize = void 0;
var serialize_1 = require("./serialize");
Object.defineProperty(exports, "serialize", { enumerable: true, get: function () { return serialize_1.serialize; } });
var deserialize_1 = require("./deserialize");
Object.defineProperty(exports, "deserialize", { enumerable: true, get: function () { return deserialize_1.deserialize; } });
Object.defineProperty(exports, "deserializeFile", { enumerable: true, get: function () { return deserialize_1.deserializeFile; } });
function isKvObject(value) {
    return typeof value !== "string";
}
exports.isKvObject = isKvObject;
function isDuplicateKeyArray(array) {
    return array._duplicateKeyBrand !== undefined;
}
exports.isDuplicateKeyArray = isDuplicateKeyArray;
function arrayToKvObject(array) {
    const obj = {};
    for (let i = 0; i < array.length; i++) {
        obj[(i + 1).toString()] = array[i];
    }
    return obj;
}
exports.arrayToKvObject = arrayToKvObject;
function arrayFromKvObject(kvobject) {
    const result = [];
    let index = 1;
    let indexString;
    while (kvobject[(indexString = index.toString())]) {
        result.push(kvobject[indexString]);
    }
    return result;
}
exports.arrayFromKvObject = arrayFromKvObject;
function createDuplicateKeyArray(array) {
    const result = [...array];
    result._duplicateKeyBrand = true;
    return result;
}
exports.createDuplicateKeyArray = createDuplicateKeyArray;
